Bring me the coke from the cabinet fails

Bring me the coke from the cabinet:

    [INFO] [1510689517.773973]: Starting executing of action Find.
    [ERROR] [1510689517.779948]: Exception in your execute callback: 'Find' object has no attribute '_find_sms'
    Traceback (most recent call last):
      File "/opt/ros/kinetic/lib/python2.7/dist-packages/actionlib/simple_action_server.py", line 289, in executeLoop
        self.execute_callback(goal)
    File "/home/amigo/ros/kinetic/repos/https_/github.com/tue-robotics/action_server.git/action_server/src/  action_server/server.py", line 71, in _add_action_cb
        action_result = self._task_manager.execute_next_action()
    File "/home/amigo/ros/kinetic/repos/https_/github.com/tue-robotics/action_server.git/action_server/src/  action_server/task_manager.py", line 69, in execute_next_action
        result = self._active_action.start()
    File "/home/amigo/ros/kinetic/repos/https_/github.com/tue-robotics/action_server.git/action_server/src/  action_server/actions/action.py", line 111, in start
        self._start()
    File "/home/amigo/ros/kinetic/repos/https_/github.com/tue-robotics/action_server.git/action_server/src/  action_server/actions/bring.py", line 183, in _start
        find_result = self._find_action.start()
    File "/home/amigo/ros/kinetic/repos/https_/github.com/tue-robotics/action_server.git/action_server/src/  action_server/actions/action.py", line 111, in start
        self._start()
    File "/home/amigo/ros/kinetic/repos/https_/github.com/tue-robotics/action_server.git/action_server/src/  action_server/actions/find.py", line 142, in _start
        for fsm in self._find_sms:
    AttributeError: 'Find' object has no attribute '_find_sms'

Complete reboot of the robot. Nu problems

```
Nov 14 22:33:55 amigo2 sudo[924]: [INFO] [1510695235.862643]: Configuring action Bring with semantics {'action': 'bring', 'source-location': {'id': 'cabinet'}, 'object': {'type': 'coke'}, 'target-location': {'type': 'person', 'id': 'operator'}} and knowledge {}.
Nov 14 22:33:55 amigo2 sudo[924]: [INFO] [1510695235.863308]: Configuring action Find with semantics {'object': {'type': 'coke'}, 'location': {'id': 'cabinet'}} and knowledge {}.
Nov 14 22:33:55 amigo2 sudo[924]: Setting up state machine with loc = cabinet, area = on_top_of, nav_area = in_front_of
Nov 14 22:33:55 amigo2 sudo[924]: [INFO] [1510695235.877953]: Resulting knowledge = {'object-designator': Designator(resolve_type=<class 'robot_skills.util.entity.Entity'>, name=None)}
Nov 14 22:33:59 amigo2 sudo[924]: [INFO] [1510695235.878787]: Configuring action PickUp with semantics {'objeLaser Pose= 0.721792 -3.26783 -1.51269
Nov 14 22:33:59 amigo2 sudo[924]: Registering Scans:Done
Nov 14 22:34:05 amigo2 sudo[924]: Succesfully stored '/home/amigo/ed/kinect/2017-11-14/on_top_of_cabinet/2017-11-14-22-34-04_rgb.png'
Nov 14 22:34:12 amigo2 sudo[924]: Laser Pose= 0.871767 -3.42931 -1.03389
Nov 14 22:34:13 amigo2 sudo[924]: Registering Scans:Done
Nov 14 22:34:14 amigo2 sudo[924]: ct': {'type': 'coke'}} and knowledge {'object-designator': Designator(resolve_type=<class 'robot_skills.util.entity.Entity'>, name=None)}.
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695235.879731]: Resulting knowledge = {'arm-designator': LockingDesignator(None), 'object-designator': Designator(resolve_type=<class 'robot_skills.util.entity.Entity'>, name=None)}
Nov 14 22:34:14 amigo2 sudo[924]: person
Nov 14 22:34:14 amigo2 sudo[924]: operator
Nov 14 22:34:14 amigo2 sudo[924]: None
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695235.888857]: Configuring action NavigateTo with semantics {'object': {'type': 'person', 'id': 'operator'}} and knowledge {}.
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695235.944968]: Navigation set up for a waypoint
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695235.945567]: Resulting knowledge = {'location-designator': Designator(resolve_type=<class 'robot_skills.util.entity.Entity'>, name=None)}
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695235.945870]: Resulting knowledge = {}
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695235.946343]: Starting executing of action Bring.
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695235.946865]: 'Bring the action!'
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695236.992396]: Starting executing of action Find.
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695236.992813]: State machine starting in initial state 'INSPECT_SOURCE_ENTITY' with userdata:
Nov 14 22:34:14 amigo2 sudo[924]:         []
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695236.993167]: State machine starting in initial state 'NAVIGATE_TO_INSPECT' with userdata:
Nov 14 22:34:14 amigo2 sudo[924]:         []
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695236.993440]: State machine starting in initial state 'START_ANALYSIS' with userdata:
Nov 14 22:34:14 amigo2 sudo[924]:         []
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695237.039041]: State machine transitioning 'START_ANALYSIS':'done'-->'NAVIGATE'
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695237.039476]: State machine starting in initial state 'GET_PLAN' with userdata:
Nov 14 22:34:14 amigo2 sudo[924]:         []
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695237.170913]: choosing best entity from this list (name->weight):
Nov 14 22:34:14 amigo2 sudo[924]:         [('cabinet', 0)]
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695237.316077]: choosing best entity from this list (name->weight):
Nov 14 22:34:14 amigo2 sudo[924]:         [('cabinet', 0)]
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695237.798383]: 'Getting there!'
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695237.812687]: State machine transitioning 'GET_PLAN':'goal_ok'-->'EXECUTE_PLAN'
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695242.221749]: State machine terminating 'EXECUTE_PLAN':'succeeded':'arrived'
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695242.222243]: State machine transitioning 'NAVIGATE':'arrived'-->'STOP_ANALYSIS_SUCCEED'
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695242.238581]: State machine terminating 'STOP_ANALYSIS_SUCCEED':'done':'arrived'
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695242.239094]: State machine transitioning 'NAVIGATE_TO_INSPECT':'arrived'-->'SEGMENT'
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695242.270425]: choosing best entity from this list (name->weight):
Nov 14 22:34:14 amigo2 sudo[924]:         [('cabinet', 0)]
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695245.587738]: >> Segmented 5 objects!
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695249.030449]:    - Object 0 is a 'sponge' (ID: 65e5318b6cd930a6b5a3080821e9899e)
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695249.030852]:    - Object 1 is a 'orange' (ID: e842ace08bacc528bdbb5c38d216bb5a)
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695249.031118]:    - Object 2 is a 'tea' (ID: 9b066111ef9079c3cdf524ebbebbfb48)
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695249.031386]:    - Object 3 is a 'hair_gel' (ID: 356d20d5b8d04fb7939bbc4f99f86389)
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695249.031714]:    - Object 4 is a 'coke' (ID: 68cf6382c9aff9bfeda16cf3b692e004)
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695249.032349]: State machine terminating 'SEGMENT':'done':'done'
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695249.032871]: State machine transitioning 'INSPECT_SOURCE_ENTITY':'done'-->'CHECK_IF_ENTITY_FOUND'
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695249.092397]: State machine terminating 'CHECK_IF_ENTITY_FOUND':'succeeded':'succeeded'
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695249.092920]: 'Hey, I found a coke!'
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695250.794974]: Starting executing of action PickUp.
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695250.795603]: State machine starting in initial state 'PREPARE_GRASP' with userdata:
Nov 14 22:34:14 amigo2 sudo[924]:         []
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695250.809177]: State machine transitioning 'PREPARE_GRASP':'succeeded'-->'NAVIGATE_TO_GRAB'
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695250.810251]: State machine starting in initial state 'START_ANALYSIS' with userdata:
Nov 14 22:34:14 amigo2 sudo[924]:         []
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695250.836211]: State machine transitioning 'START_ANALYSIS':'done'-->'NAVIGATE'
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695250.836616]: State machine starting in initial state 'GET_PLAN' with userdata:
Nov 14 22:34:14 amigo2 sudo[924]:         []
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695250.937281]: Navigating to grasp entity id:68cf6382c9aff9bfeda16cf3b692e004
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695251.314067]: 'I will be there in a sec!'
Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695251.370424]: State machine transitioning 'GET_PLAN':'goal_ok'-->'EXECUTE_PLAN'
Nov 14 22:34:19 amigo2 sudo[924]: [INFO] [1510695254.176204]: State machineSuccesfully stored '/home/amigo/ed/kinect/2017-11-14/68cf6382c9aff9bfeda16cf3b692e004/2017-11-14-22-34-18_rgb.png'
Nov 14 22:34:32 amigo2 sudo[924]: [WARN] [1510695272.489158]: Grasp-precompute error too large: [-0.00470041719912, -0.00599204815735, -0.00401351494319]
Nov 14 22:34:34 amigo2 sudo[924]: Succesfully stored '/home/amigo/ed/kinect/2017-11-14/68cf6382c9aff9bfeda16cf3b692e004/2017-11-14-22-34-34_rgb.png'
Nov 14 22:34:55 amigo2 sudo[924]: [ERROR] [1510695295.957538]: timeout exceeded while waiting for message on topic /amigo/handoverdetector_right/result
Nov 14 22:35:05 amigo2 sudo[924]: [ERROR] [1510695305.982313]: timeout exceeded while waiting for message on topic /amigo/handoverdetector_right/result
Nov 14 22:35:08 amigo2 sudo[924]: [ERROR] [1510695308.517176]: send_gripper_goal open is called but there is still an entity with id '68cf6382c9aff9bfeda16cf3b692e004' occupying the gripper, please update the world model and remove this entity
```

This is already fixed :)

2017-11-14 22:36 GMT+01:00 Matthijs van der Burgh <notifications@github.com>
:

> Complete reboot of the robot. Nu problems
>
> Nov 14 22:33:55 amigo2 sudo[924]: [INFO] [1510695235.862643]: Configuring action Bring with semantics {'action': 'bring', 'source-location': {'id': 'cabinet'}, 'object': {'type': 'coke'}, 'target-location': {'type': 'person', 'id': 'operator'}} and knowledge {}.
> Nov 14 22:33:55 amigo2 sudo[924]: [INFO] [1510695235.863308]: Configuring action Find with semantics {'object': {'type': 'coke'}, 'location': {'id': 'cabinet'}} and knowledge {}.
> Nov 14 22:33:55 amigo2 sudo[924]: Setting up state machine with loc = cabinet, area = on_top_of, nav_area = in_front_of
> Nov 14 22:33:55 amigo2 sudo[924]: [INFO] [1510695235.877953]: Resulting knowledge = {'object-designator': Designator(resolve_type=<class 'robot_skills.util.entity.Entity'>, name=None)}
> Nov 14 22:33:59 amigo2 sudo[924]: [INFO] [1510695235.878787]: Configuring action PickUp with semantics {'objeLaser Pose= 0.721792 -3.26783 -1.51269
> Nov 14 22:33:59 amigo2 sudo[924]: Registering Scans:Done
> Nov 14 22:34:05 amigo2 sudo[924]: Succesfully stored '/home/amigo/ed/kinect/2017-11-14/on_top_of_cabinet/2017-11-14-22-34-04_rgb.png'
> Nov 14 22:34:12 amigo2 sudo[924]: Laser Pose= 0.871767 -3.42931 -1.03389
> Nov 14 22:34:13 amigo2 sudo[924]: Registering Scans:Done
> Nov 14 22:34:14 amigo2 sudo[924]: ct': {'type': 'coke'}} and knowledge {'object-designator': Designator(resolve_type=<class 'robot_skills.util.entity.Entity'>, name=None)}.
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695235.879731]: Resulting knowledge = {'arm-designator': LockingDesignator(None), 'object-designator': Designator(resolve_type=<class 'robot_skills.util.entity.Entity'>, name=None)}
> Nov 14 22:34:14 amigo2 sudo[924]: person
> Nov 14 22:34:14 amigo2 sudo[924]: operator
> Nov 14 22:34:14 amigo2 sudo[924]: None
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695235.888857]: Configuring action NavigateTo with semantics {'object': {'type': 'person', 'id': 'operator'}} and knowledge {}.
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695235.944968]: Navigation set up for a waypoint
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695235.945567]: Resulting knowledge = {'location-designator': Designator(resolve_type=<class 'robot_skills.util.entity.Entity'>, name=None)}
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695235.945870]: Resulting knowledge = {}
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695235.946343]: Starting executing of action Bring.
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695235.946865]: 'Bring the action!'
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695236.992396]: Starting executing of action Find.
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695236.992813]: State machine starting in initial state 'INSPECT_SOURCE_ENTITY' with userdata:
> Nov 14 22:34:14 amigo2 sudo[924]:         []
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695236.993167]: State machine starting in initial state 'NAVIGATE_TO_INSPECT' with userdata:
> Nov 14 22:34:14 amigo2 sudo[924]:         []
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695236.993440]: State machine starting in initial state 'START_ANALYSIS' with userdata:
> Nov 14 22:34:14 amigo2 sudo[924]:         []
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695237.039041]: State machine transitioning 'START_ANALYSIS':'done'-->'NAVIGATE'
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695237.039476]: State machine starting in initial state 'GET_PLAN' with userdata:
> Nov 14 22:34:14 amigo2 sudo[924]:         []
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695237.170913]: choosing best entity from this list (name->weight):
> Nov 14 22:34:14 amigo2 sudo[924]:         [('cabinet', 0)]
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695237.316077]: choosing best entity from this list (name->weight):
> Nov 14 22:34:14 amigo2 sudo[924]:         [('cabinet', 0)]
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695237.798383]: 'Getting there!'
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695237.812687]: State machine transitioning 'GET_PLAN':'goal_ok'-->'EXECUTE_PLAN'
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695242.221749]: State machine terminating 'EXECUTE_PLAN':'succeeded':'arrived'
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695242.222243]: State machine transitioning 'NAVIGATE':'arrived'-->'STOP_ANALYSIS_SUCCEED'
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695242.238581]: State machine terminating 'STOP_ANALYSIS_SUCCEED':'done':'arrived'
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695242.239094]: State machine transitioning 'NAVIGATE_TO_INSPECT':'arrived'-->'SEGMENT'
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695242.270425]: choosing best entity from this list (name->weight):
> Nov 14 22:34:14 amigo2 sudo[924]:         [('cabinet', 0)]
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695245.587738]: >> Segmented 5 objects!
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695249.030449]:    - Object 0 is a 'sponge' (ID: 65e5318b6cd930a6b5a3080821e9899e)
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695249.030852]:    - Object 1 is a 'orange' (ID: e842ace08bacc528bdbb5c38d216bb5a)
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695249.031118]:    - Object 2 is a 'tea' (ID: 9b066111ef9079c3cdf524ebbebbfb48)
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695249.031386]:    - Object 3 is a 'hair_gel' (ID: 356d20d5b8d04fb7939bbc4f99f86389)
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695249.031714]:    - Object 4 is a 'coke' (ID: 68cf6382c9aff9bfeda16cf3b692e004)
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695249.032349]: State machine terminating 'SEGMENT':'done':'done'
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695249.032871]: State machine transitioning 'INSPECT_SOURCE_ENTITY':'done'-->'CHECK_IF_ENTITY_FOUND'
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695249.092397]: State machine terminating 'CHECK_IF_ENTITY_FOUND':'succeeded':'succeeded'
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695249.092920]: 'Hey, I found a coke!'
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695250.794974]: Starting executing of action PickUp.
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695250.795603]: State machine starting in initial state 'PREPARE_GRASP' with userdata:
> Nov 14 22:34:14 amigo2 sudo[924]:         []
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695250.809177]: State machine transitioning 'PREPARE_GRASP':'succeeded'-->'NAVIGATE_TO_GRAB'
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695250.810251]: State machine starting in initial state 'START_ANALYSIS' with userdata:
> Nov 14 22:34:14 amigo2 sudo[924]:         []
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695250.836211]: State machine transitioning 'START_ANALYSIS':'done'-->'NAVIGATE'
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695250.836616]: State machine starting in initial state 'GET_PLAN' with userdata:
> Nov 14 22:34:14 amigo2 sudo[924]:         []
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695250.937281]: Navigating to grasp entity id:68cf6382c9aff9bfeda16cf3b692e004
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695251.314067]: 'I will be there in a sec!'
> Nov 14 22:34:14 amigo2 sudo[924]: [INFO] [1510695251.370424]: State machine transitioning 'GET_PLAN':'goal_ok'-->'EXECUTE_PLAN'
> Nov 14 22:34:19 amigo2 sudo[924]: [INFO] [1510695254.176204]: State machineSuccesfully stored '/home/amigo/ed/kinect/2017-11-14/68cf6382c9aff9bfeda16cf3b692e004/2017-11-14-22-34-18_rgb.png'
> Nov 14 22:34:32 amigo2 sudo[924]: [WARN] [1510695272.489158]: Grasp-precompute error too large: [-0.00470041719912, -0.00599204815735, -0.00401351494319]
> Nov 14 22:34:34 amigo2 sudo[924]: Succesfully stored '/home/amigo/ed/kinect/2017-11-14/68cf6382c9aff9bfeda16cf3b692e004/2017-11-14-22-34-34_rgb.png'
> Nov 14 22:34:55 amigo2 sudo[924]: [ERROR] [1510695295.957538]: timeout exceeded while waiting for message on topic /amigo/handoverdetector_right/result
> Nov 14 22:35:05 amigo2 sudo[924]: [ERROR] [1510695305.982313]: timeout exceeded while waiting for message on topic /amigo/handoverdetector_right/result
> Nov 14 22:35:08 amigo2 sudo[924]: [ERROR] [1510695308.517176]: send_gripper_goal open is called but there is still an entity with id '68cf6382c9aff9bfeda16cf3b692e004' occupying the gripper, please update the world model and remove this entity
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/tue-robotics/action_server/issues/85#issuecomment-344406942>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIlLByVoF7DI_0d9bZmAt1Eno4i-U0sYks5s2gfbgaJpZM4Qd6Lv>
> .
>

--- a/action_server/src/action_server/actions/find.py
+++ b/action_server/src/action_server/actions/find.py
@@ -115,7 +115,7 @@ class Find(Action):
                 navigation_area_designator = VariableDesignator(nav_area)
 
                 # Set up the Find state machine
-                print "Setting up state machine with loc = {}, area = {}, nav_area = {}".format(loc, area, nav_area)
+                rospy.loginfo("Setting up state machine with loc = {}, area = {}, nav_area = {}".format(loc, area, nav_area))
                 self._find_state_machines.append(states.Find(robot=self._robot,
                                                              source_entity_designator=location_designator,
                                                              description_designator=description_designator,

--- a/action_server/src/action_server/actions/find.py
+++ b/action_server/src/action_server/actions/find.py
@@ -102,31 +102,26 @@ class Find(Action):
         entity_description = {'type': self._object.type}
         description_designator = VariableDesignator(entity_description)
 
-        # If we can already find an entity using the current description, use that
-        (entities, error_msg) = entities_from_description(entity_description, robot)
-        if entities:
-            self._found_entity_designator = EdEntityDesignator(id=entities[0].id, robot=robot)
-        else:
-            # Set up designator to be filled with the found entity
-            self._found_entity_designator = VariableDesignator(resolve_type=Entity)
+        # Set up designator to be filled with the found entity
+        self._found_entity_designator = VariableDesignator(resolve_type=Entity)
 
-            self._find_sms = []
-            for loc, areas in self._areas.iteritems():
-                location_designator = EdEntityDesignator(self._robot, id=loc)
-                nav_area = self._nav_areas[loc]
-                for area in areas:
-                    area_designator = VariableDesignator(area)
+        self._find_state_machines = []
+        for loc, areas in self._areas.iteritems():
+            location_designator = EdEntityDesignator(self._robot, id=loc)
+            nav_area = self._nav_areas[loc]
+            for area in areas:
+                area_designator = VariableDesignator(area)
 
-                    navigation_area_designator = VariableDesignator(nav_area)
+                navigation_area_designator = VariableDesignator(nav_area)
 
-                    # Set up the Find state machine
-                    print "Setting up state machine with loc = {}, area = {}, nav_area = {}".format(loc, area, nav_area)
-                    self._find_sms.append(states.Find(robot=self._robot,
-                                                      source_entity_designator=location_designator,
-                                                      description_designator=description_designator,
-                                                      area_name_designator=area_designator,
-                                                      navigation_area_designator=navigation_area_designator,
-                                                      found_entity_designator=self._found_entity_designator))
+                # Set up the Find state machine
+                print "Setting up state machine with loc = {}, area = {}, nav_area = {}".format(loc, area, nav_area)
+                self._find_state_machines.append(states.Find(robot=self._robot,
+                                                             source_entity_designator=location_designator,
+                                                             description_designator=description_designator,
+                                                             area_name_designator=area_designator,
+                                                             navigation_area_designator=navigation_area_designator,
+                                                             found_entity_designator=self._found_entity_designator))
 
         self._config_result.resulting_knowledge['object-designator'] = self._found_entity_designator
         self._config_result.succeeded = True
@@ -139,7 +134,7 @@ class Find(Action):
         #     self._execute_result.succeeded = True
         #     return
 
-        for fsm in self._find_sms:
+        for fsm in self._find_state_machines:
             res = fsm.execute()
 
             if res == 'succeeded':

